File: Purok Management System/UI/PMS-UI/src/app/dto/command/createperson.dto.ts
import { CreateAddressDto } from "./createaddress.dto";
import { CreateHealthBackgroundDto } from "./createhealthbackground.dto";

export class CreatePersonDto {
    personId?:number;
    firstName?: string;
    lastName?: string;
    middleName?: string;
    birthPlace?: string;
    birthDate?: Date | string;
    telephoneNo?: string;
    mobileNo?: string;
    emailAddress?: string;
    height?: number;
    weight?: number;
    religion?: string;
    withNatId?: boolean;
    withSssGsis?: boolean;
    withDriversLicense?: boolean;
    isRegisteredVoter?: boolean;
    precintNo?: string;
    photoLink?: string;
    isActive?: boolean;
    genderId?: number;
    bloodTypeId?: number;
    civilStatusId?: number;
    citizenshipId?: number;
    healthBackground?: CreateHealthBackgroundDto;
    address?: CreateAddressDto;
}

File: Purok Management System/API/PMS-API/DTO/Person/CreatePersonDto.cs

using PMS_API.DTO.Address;
using PMS_API.DTO.HealthBackground;
using PMS_API.Models;
using System.ComponentModel.DataAnnotations;

namespace PMS_API.DTO.PersonDtos
{
    public class CreatePersonDto
    {
        [Required(ErrorMessage = "First Name cannot be empty.")]
        public string FirstName { get; set; }
        [Required(ErrorMessage = "Last Name cannot be empty.")]
        public string LastName { get; set; }
        public string MiddleName { get; set; }
        public string BirthPlace { get; set; }
        public DateTime BirthDate { get; set; }
        public string TelephoneNo { get; set; }
        [Phone]
        public string MobileNo { get; set; }
        [EmailAddress]
        public string EmailAddress { get; set; }
        public float Height { get; set; }
        public float Weight { get; set; }
        public string Religion { get; set; }
        public bool WithNatId { get; set; }
        public bool WithSssGsis { get; set; }
        public bool WithDriversLicense { get; set; }
        public bool IsRegisteredVoter { get; set; }
        public string PrecintNo { get; set; }
        public string PhotoLink { get; set; }
        public bool IsActive { get; set; }
        [Required(ErrorMessage = " Please select your gender")]
        public int GenderId { get; set; }
        public int BloodTypeId { get; set; }
        public int CivilStatusId { get; set; }
        public int CitizenshipId { get; set; }
        public CreateHealthBackgroundDto HealthBackground { get; set; }
        public CreateAddressDto Address { get; set; }
    }
}
